csharp/src/lesson03/README.md

```diff
@@ -0,0 +1,327 @@
+```csharp
+[HttpGet("{helloString}", Name = "GetFormat")]
+public string Get(string helloString)
+{
+    var headers = Request.Headers.ToDictionary(k => k.Key, v => v.Value.First());
+    using (var scope = StartServerSpan(_tracer, headers, "format-controller"))
+    {
+        var formattedHelloString = $"Hello, {helloString}!";
+        scope.Span.Log(new Dictionary<string, object>
+        {
+            [LogFields.Event] = "string-format",
+            ["value"] = formattedHelloString
+        });
+        return formattedHelloString;
+    }
+}
+```
+
+### Take It For a Spin
+
+As before, first run the `Lesson03.Exercise.Server` project in Visual Studio.
+Then run `Lesson03.Exercise.Client` in the terminal. You should see an output like this:
+
+Server output (ASP.NET Core Web Server):
+
+```powershell
+$ dotnet run
+info: Jaeger.Configuration[0]
+      Initialized Jaeger.Tracer
+Hosting environment: Development
+Content root path: opentracing-tutorial\csharp\src\lesson03\example\Lesson03.Example.Server
+Now listening on: http://localhost:8081
+Application started. Press Ctrl+C to shut down.
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:1924ee38b5e4b7c3:52a7512a6a2d7ccc:1 - format-controller
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:291f14c7a20258ba:52a7512a6a2d7ccc:1 - Result ObjectResult
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:52a7512a6a2d7ccc:1e05b0ca172ad56e:1 - Action OpenTracing.Tutorial.Lesson03.Example.Server.Controllers.FormatController/Get
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:1e05b0ca172ad56e:92ffe8a50d62d2e:1 - HTTP GET
+```
+
+Client terminal output:
+
+```powershell
+$ dotnet run Bryan
+info: Jaeger.Configuration[0]
+      Initialized Jaeger.Tracer
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:92ffe8a50d62d2e:58eee39b348263cc:1 - format-string
+info: OpenTracing.Tutorial.Lesson03.Example.Client.Hello[0]
+      Hello, Bryan!
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:2dc30af91f18b434:58eee39b348263cc:1 - print-hello
+info: Jaeger.Reporters.LoggingReporter[0]
+      Span reported: 722582a9299820e258eee39b348263cc:58eee39b348263cc:0:1 - say-hello
+```
+
+Note how all recorded spans show the same trace ID `722582a9299820e258eee39b348263cc`.
+This is a sign of correct instrumentation. It is also a very useful debugging approach when something
+is wrong with tracing. A typical error is to miss the context propagation somwehere,
+either in-process or inter-process, which results in different trace IDs and broken
+traces.
+
+If we open this trace in the UI, we see all spans across client _and_ server.
+
+![Trace](trace-server-unfinished.png)
+
+### Instrumenting the Server using `OpenTracing.Contrib.NetCore`
```

Comment:
this was mentioned as extra credit

Reply:
Good catch. We first had it only as extra and afterwards decided to add it to the tutorial without removing it there. Didn’t even thought about it while adding OpenTracing.Contrib.Grpc yesterday...